src/humanPose/AccelerationLens.js

```diff
@@ -32,34 +35,59 @@ export class AccelerationLens extends MotionStudyLens {
      * @return {boolean} True if the joint has had its acceleration calculated, false otherwise.
      */
     accelerationAppliedToJoint(joint) {
-        return joint.acceleration && joint.accelerationMagnitude > 0.000001
+        return joint.acceleration && joint.accelerationMagnitude && joint.accelerationMagnitude !== -1
     }
     
     applyLensToPose(pose) {
         const previousPose = pose.metadata.previousPose;
         const previousPreviousPose = previousPose ? previousPose.metadata.previousPose : null;
         if (!previousPose) {
             pose.forEachJoint(joint => {
-                joint.velocity = new THREE.Vector3(); // Velocity is zero for the first pose
-                joint.speed = 0;
-                joint.acceleration = new THREE.Vector3(); // Acceleration is zero for the first two poses
-                joint.accelerationMagnitude = 0;
+                // Velocity and acceleration are zero for the first pose
+                joint.velocity = new THREE.Vector3(); 
+                joint.speed = -1; // -1 means that the metric cannot calculated based on data
+                joint.acceleration = new THREE.Vector3();
+                joint.accelerationMagnitude = -1; // -1 means that the metric cannot calculated based on data
             });
         } else if (!previousPreviousPose) {
             pose.forEachJoint(joint => {
                 const previousJoint = previousPose.getJoint(joint.name);
-                joint.velocity = joint.position.clone().sub(previousJoint.position).divideScalar(pose.timestamp - previousPose.timestamp); // mm/ms = m/s
-                joint.speed = joint.velocity.length();
-                joint.acceleration = new THREE.Vector3(); // Acceleration is zero for the first two poses
-                joint.accelerationMagnitude = 0;
+                if (!joint.valid || !previousJoint.valid) {
+                    // Velocity and acceleration are zero if a joint is invalid in current or previous pose
+                    joint.velocity = new THREE.Vector3();
+                    joint.speed = -1;
```

Comment:
Kind of a nit, but I wonder if it would be better to use null instead of -1. ~~Otherwise, someone writing code around this in the future might not realize to check for -1 and do some math which will work because -1 is a number.~~ Actually, js casts null to 0 when doing math with it, so this point is moot. I have a preference for null instead of -1, but this is really just a nit, feel free to ignore.

Reply:
It's super minor, but any use of null causes deoptimization from the value's slot having to be either a number or null